extension/extensionhelper/factory.go

```diff
@@ -15,60 +15,29 @@
 package extensionhelper // import "go.opentelemetry.io/collector/extension/extensionhelper"
 
 import (
-	"context"
-
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/config"
-	"go.opentelemetry.io/collector/internal/internalinterface"
 )
 
-// FactoryOption apply changes to ExporterOptions.
-type FactoryOption func(o *factory)
-
-// CreateDefaultConfig is the equivalent of component.ExtensionFactory.CreateDefaultConfig()
-type CreateDefaultConfig func() config.Extension
+// Deprecated: not needed, will be removed soon.
+type FactoryOption struct{}
 
-// CreateServiceExtension is the equivalent of component.ExtensionFactory.CreateExtension()
-type CreateServiceExtension func(context.Context, component.ExtensionCreateSettings, config.Extension) (component.Extension, error)
+// Deprecated: use component.ExtensionCreateDefaultConfigFunc.
+type CreateDefaultConfig = component.ExtensionCreateDefaultConfigFunc
 
-type factory struct {
-	internalinterface.BaseInternal
-	cfgType                config.Type
-	createDefaultConfig    CreateDefaultConfig
-	createServiceExtension CreateServiceExtension
-}
+// Deprecated: use component.CreateExtensionFunc.
+type CreateServiceExtension = component.CreateExtensionFunc
 
-// NewFactory returns a component.ExtensionFactory.
+// Deprecated: use component.ExtensionFactory directly.
```

Comment:
Naturally the return type of this should be `*component.ExtensionFactory` but that will be a breaking change, also the component.Factories also uses the non pointers. Maybe we should keep the interface and have the same trick with not allowing extension?

Reply:
If we have an interface, why wouldn't we allow people to extend it?

Reply:
> If we have an interface, why wouldn't we allow people to extend it?

If we add a new signal type (say, profiles), how would you add a new method to create a profiles exporter/receiver/processor?

Reply:
This reasoning should definitely be documented.